src/core/domain.py

```diff
@@ -94,6 +94,65 @@ def log_dir(self) -> str:
         return f"s3a://{self.bucket}/{self.path}"
 
 
+class AzureStorageConnectionInfo:
+    """Class representing credentials and endpoints to connect to S3."""
+
+    def __init__(self, relation_data):
+        self.relation_data = relation_data
+
+    @property
+    def endpoint(self) -> str | None:
+        """Return endpoint of the Azure storage container."""
+        if self.connection_protocol in ("abfs", "abfss"):
+            return f"{self.connection_protocol}:/{self.container}@{self.storage_account}.dfs.core.windows.net"
+        elif self.connection_protocol in ("wasb", "wasbs"):
+            return f"{self.connection_protocol}:/{self.container}@{self.storage_account}.blob.core.windows.net"
+        return ""
+
+    @property
+    def secret_key(self) -> str:
+        """Return the secret key."""
+        return self.relation_data["secret-key"]
+
+    @property
+    def path(self) -> str:
+        """Return the path in the Azure Storage container."""
+        return self.relation_data.get("path", "")
+
+    @property
+    def container(self) -> str:
+        """Return the name of the Azure Storage container."""
+        return self.relation_data["container"]
+
+    @property
+    def connection_protocol(self) -> str:
+        """Return the protocol to be used to access files."""
+        return self.relation_data["connection-protocol"].lower()
+
+    @property
+    def storage_account(self) -> str:
+        """Return the name of the Azure Storage account."""
+        return self.relation_data["storage-account"]
+
+    @property
+    def log_dir(self) -> str:
+        """Return the full path to the object."""
+        if self.endpoint:
+            return f"{self.endpoint}/{self.path}"
+        return ""
+
+    @property
+    def connection_string(self) -> str:
```

Comment:
**question** I tried a simple search and replace in the PR here, but where is this used? For what do we need this format?

Reply:
I added this while writing the `verify` function for the Azure Storage Manager class. A client needs this connection string to connect to the Azure Storage. For now, we have removed that validation because it was taking too long. This function does nothing now, but will be useful when we add the credential validation later.

Reply:
**thought, suggestion** If it is not used I would probably remove it for keeping things clean. Also the format needs rather special, maybe this conversion should eventually be in the manager, rather than the domain object.

Reply:
Removed the property from here.